docs/Userguide_sharing_data.rst

```diff
@@ -0,0 +1,101 @@
+Sharing Data with ACLs
+======================
+
+As an illustrative example, to allow ``$USER`` to share with ``$USER2`` in
+``rwx`` a hierarchy ``/network/scratch/$USER/X/Y/Z/...`` with ACLs:
+
+----
+
+| Grant **oneself** permissions to access any **future** files/folders created
+  by the other *(or oneself)*
+| (``-d`` renders this permission a "default" / inheritable one)
+
+.. code-block:: bash
+
+   setfacl -Rdm user:$USER:rwx  /network/scratch/$USER/X/Y/Z/
+
+----
+
+| Grant **another** permission to access any **future** files/folders created
+  by the other *(or oneself)*
+| (``-d`` renders this permission a "default" / inheritable one)
+
+.. code-block:: bash
+
+   setfacl -Rdm user:$USER2:rwx /network/scratch/$USER/X/Y/Z/
+
+----
+
+| Grant **another** permission to access any **existing** files/folders created
+  by *oneself*.
+| Such files and folders were created before the new default ACLs were added
+  above and thus did not inherit them from their parent folder at the moment of
+  their creation.
+
+.. code-block:: bash
+
+   setfacl -Rm  user:$USER2:rwx /network/scratch/$USER/X/Y/Z/
+
+----
+
+| Grant **another** permission to search through one's hierarchy down to the
+  shared location in question.
```

Comment:
Why would we want to do that? What does "search" mean? Only `ls`? What about reading?

Reply:
To put it in extremely plain language, Unix separates the concept of "traversing" a directory from "reading" it.

To execute `ls /network/scratch/$USER/a/b/c/workspace/`, a `$USER2` **does not** require "read" (`r`) permission on _anything_ except `workspace/` itself. A `$USER2` **does** however require "search" permissions on **all** directories from the root down to `workspace/`. Unix overloads the "executable" (`x`) permission (which makes no sense on directories) for "search".

The rules are somewhat baroque but are elaborated upon in greater detail in `man path_resolution`. The summary that's important for most people is what I wrote above, and what my proposed `setfacl` commands arrange for.